src/scala/ly/stealth/mesos/kafka/Scheduler.scala

```diff
@@ -87,8 +87,14 @@ object Scheduler extends org.apache.mesos.Scheduler {
         "host.name" -> offer.getHostname
       )
 
-      if (HttpServer.kafkaVersion.compareTo(new Version("0.9")) >= 0)
-        defaults += ("listeners" -> s"PLAINTEXT://:${reservation.port}")
+      if (HttpServer.kafkaVersion.compareTo(new Version("0.9")) >= 0) {
+        if (broker.bindAddress != null) {
+          defaults += ("listeners" -> s"PLAINTEXT://${offer.getHostname}:${reservation.port}")
```

Comment:
@steveniemitz 
I tried to add broker as below and then started,
./kafka-mesos.sh broker add 0 --heap 256 --mem 512 --cpu 0.2 --bind-address if:eth1

Code mentioned above is in Scheduler.newTask(). This function is invoked in driver after offer is accepted. So here, bindAddress.resolve() will return IP address of driver rather than executor. 
The call stack for "--bind-address" scenario is almost like below,
 at ly.stealth.mesos.kafka.Util$BindAddress.resolveInterfaceAddress(Util.scala:111)
        at ly.stealth.mesos.kafka.Util$BindAddress.resolve(Util.scala:92)
        at ly.stealth.mesos.kafka.Scheduler$.taskData$1(Scheduler.scala:92)
	at ly.stealth.mesos.kafka.Scheduler$.newTask(Scheduler.scala:113)
	at ly.stealth.mesos.kafka.Scheduler$.launchTask(Scheduler.scala:318)
	at ly.stealth.mesos.kafka.Scheduler$$anonfun$acceptOffer$2.apply(Scheduler.scala:240)
	at ly.stealth.mesos.kafka.Scheduler$$anonfun$acceptOffer$2.apply(Scheduler.scala:236)
	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:47)
	at ly.stealth.mesos.kafka.Scheduler$.acceptOffer(Scheduler.scala:236)
	at ly.stealth.mesos.kafka.Scheduler$$anonfun$syncBrokers$1.apply(Scheduler.scala:203)
	at ly.stealth.mesos.kafka.Scheduler$$anonfun$syncBrokers$1.apply(Scheduler.scala:202)
	at scala.collection.Iterator$class.foreach(Iterator.scala:727)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1157)
	at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at ly.stealth.mesos.kafka.Scheduler$.syncBrokers(Scheduler.scala:202)
	at ly.stealth.mesos.kafka.Scheduler$.resourceOffers(Scheduler.scala:139)

After offer is accepted, executor will resolve address based on "--bind-address" on mesos agent node. If everything goes well, broker will be started.
Back to the test case I ran, when the agent did not have an interface named "eth1", it failed with following exceptions,
java.lang.IllegalStateException: Failed to resolve if:eth1
	at ly.stealth.mesos.kafka.Util$BindAddress.resolveInterfaceAddress(Util.scala:111)
	at ly.stealth.mesos.kafka.Util$BindAddress.resolve(Util.scala:92)
	at ly.stealth.mesos.kafka.Broker.options(Broker.scala:71)
	at ly.stealth.mesos.kafka.KafkaServer.start(BrokerServer.scala:50)
	at ly.stealth.mesos.kafka.Executor$.ly$stealth$mesos$kafka$Executor$$runBroker0$1(Executor.scala:89)
	at ly.stealth.mesos.kafka.Executor$$anon$1.run(Executor.scala:112)